### ISSUE TITLE
new update danger zone not working. what is the offset now?

### ISSUE THREAD

**Issue by ghost** [iseekwonderful/csgoGlow#263]

**Comment by macacodobem** [comment]
For me too, don't working after danger zone updating.
need change 0xAC34; I think

**Comment by itsfrank115** [comment]
omg same do u guys have any clue of what to do to fix it?

**Comment by itsfrank115** [comment]
did anybody find the new offset?

**Comment by jefcostaa** [comment]
Anything work here, I tried 00 to 100 but nothing

**Comment by diogosilva212** [comment]
For PC is 0xA3F8, maybe someone can "convert"...

**Comment by jefcostaa** [comment]
the new is 0xA38

**Comment by macacodobem** [comment]
not working 0xAC38

**Comment by itsfrank115** [comment]
yea me too 0xAC38 isn't working for me. what should i do?

**Comment by diogosilva212** [comment]
@imoplex broooo!

**Comment by macacodobem** [comment]
@aKalisch where are you?

**Comment by macacodobem** [comment]
@jefcostaa Are sure u tried 00 to 99? or another fake news? because 0xAC38 not working

**Comment by diogosilva212** [comment]
We just need someone who can adjust the "0xA3F8" to mac...

**Comment by macacodobem** [comment]
wrong

**Comment by ethano8225** [comment]
lol you cant really just "convert" it to mac. the dude tried 38 but it didnt work lol he even said that

**Comment by diogosilva212** [comment]
On the previous update the guy said "Computed by looking at the difference in iGlowIndex offset between previous and this update." And one said "in fact, I saw the last 3 changes. realized the variable was being incremented by 16 (in hexadecimal)"

**Comment by itsfrank115** [comment]
so the next one shouldnt be 0xAC38?

**Comment by diogosilva212** [comment]
I don't know... I tried to post on "unknowncheats" forum, but they didn't approve my comment lol

**Comment by itsfrank115** [comment]
how did we figure it out for the last update?

**Comment by itsfrank115** [comment]
can't we just do it again with the same method  for this one?

**Comment by diogosilva212** [comment]
Someone posted it and said "Computed by looking at the difference in iGlowIndex offset between previous and this update" xD

**Comment by diogosilva212** [comment]
That's the problem, we don't know the method...

**Comment by itsfrank115** [comment]
before was 0xAC34 and the one before that was 0xAC10

**Comment by ethano8225** [comment]
well
it went up 24
so try whatever that is (like 34+24, idk hexadecimal)
right? lmao

**Comment by diogosilva212** [comment]
It was 0xAC00, then 0xAC10, then 0xAC20 and then 0xAC34...

**Comment by itsfrank115** [comment]
so it could basically be between 34 and 100

**Comment by diogosilva212** [comment]
I'm pretty sure this is so easy for whoever understands the basics of this... I'm feeling dumb.

**Comment by diogosilva212** [comment]
But don't try all those numbers. For Pc it was 0xA310, 0xA330, 0xA344 and now is something completely different: 0xA3F8

**Comment by itsfrank115** [comment]
oh dam you're right

**Comment by itsfrank115** [comment]
can u look up how to convert it?

**Comment by diogosilva212** [comment]
I don't know nothing about this, unfortunately

**Comment by macacodobem** [comment]
Will the "unknowncheats" staff help with the problem? u got send the topic In forum?

**Comment by ethano8225** [comment]
oh then just try to look upevery pc csgo pointer for the past few years and if there is another F and we also have the numbers for that same update we should be able to guess

**Comment by diogosilva212** [comment]
They didn't approve my reply on the "Structs offsets" forum, where the pc ones where posted, don't know...

**Comment by ethano8225** [comment]
actually hold up, i think it changed by going down 21 points, we could try that (like make 34 into 13)

**Comment by diogosilva212** [comment]
Nop, didn't work.

**Comment by ethano8225** [comment]
idk just an idea... imma go to bed now lmao gotta wake up real early tmrw

**Comment by diogosilva212** [comment]
@aKalisch @imoplex @paulinelymorgan someone heeeelp

**Comment by ghost** [comment]
ill have a look into it... gimme half an hour

**Comment by diogosilva212** [comment]
I maybe have something to help. We just need the pattern scanner and then convert the decimal number to hexadecimal

**Comment by diogosilva212** [comment]
On Pc it was on the previous update 0xA344 that in decimal was 41796. Now it's 41976, that in hexadecimal it's 0xA3F8

**Comment by diogosilva212** [comment]
For mac, this is the decimal on previous updates: 0xAC00 44032; 0xAC10 44048; 0xAC20 44064; 0xAC34 44084

**Comment by ghost** [comment]
just use a for loop to iterate over...

**Comment by diogosilva212** [comment]
@fugiefire that's chinese to me bro, sorry...

**Comment by ghost** [comment]
and its not incrementing in a pattern, its dependent on what comes before in the binary.

**Comment by macacodobem** [comment]
@fugiefire come to save us!

**Comment by diogosilva212** [comment]
What I was saying was that before, on this hack, you could use this tool (https://github.com/gabsens/Pattern-scanner-for-OSX) to discover the offset. Once you had that value, you only had to convert to hexadecimal...

**Comment by diogosilva212** [comment]
Save us, please @fugiefire <3

**Comment by amcgregor** [comment]
As an open source developer (not of cheats, sadly) it is onslaught threads like this that cause me the greatest distress, and shake whatever faith in humanity I may have still possessed.  (No, I don't know the offset; have some patience, people.)

**Comment by ghost** [comment]
^ yes please can everyone have a lil patience and not tag us?

**Comment by diogosilva212** [comment]
@amcgregor dude, we're just trying to figure this out on our own. Personally, I don't know shi*. It's exactly to avoid tagging people like you that we're trying to solve this...

**Comment by ethano8225** [comment]
I mean they the smart ones right lmao

**Comment by diogosilva212** [comment]
Yup, but they weren't here before, so we tried :) Now we wait.

**Comment by ghost** [comment]
i think its either ac49 or ac96 lemme try

**Comment by ghost** [comment]
nvm its neither lol

**Comment by ghost** [comment]
tried 49 so prob 96

**Comment by ghost** [comment]
lol my b then

**Comment by ghost** [comment]
tf does strstr() do? @amcgregor

**Comment by diogosilva212** [comment]
Lmao just saw that on the previous two updates, people in here just got lucky cause the value incremented the same (decimal) xD

**Comment by ghost** [comment]
huh?

**Comment by diogosilva212** [comment]
In one update, the decimal value of "m_iGlowIndex" went from 44032 to 44048 (+16) so in the next update some guy added more 16, making it 44064 and it actually worked

**Comment by diogosilva212** [comment]
On the next one, someone saw that on the Pc value it was +20, so added +20 on 44064 and it worked again xD

**Comment by ethano8225** [comment]
maybe it’s -16? Idk how work lmao

**Comment by diogosilva212** [comment]
Here, I can't help more:

PC Values on updates:

0xA310 ---> 41744
0xA330 ---> 41776    (+32)
0xA344 ---> 41796    (+20)
(**new**) 0xA3F8 ---> 41976     (+180)

MAC Values:

0xAC00 ---> 44032
0xAC10 ---> 44048    (+16)
0xAC20 ---> 44064    (+16)
(last known) 0xAC34 ---> 44084    (+20)

**Comment by ethano8225** [comment]
or if they added pointers it because of the other gamemode it could be like +(16 times 2, 16 times 4

**Comment by ghost** [comment]
Well, I disassembled client.dylib and tried the offsets of all the data segments nearby... none worked lmao

**Comment by diogosilva212** [comment]
How does this guy gets this? https://github.com/frk1/hazedumper/blob/master/csgo.cs

**Comment by ghost** [comment]
idk

**Comment by ghost** [comment]
oh its for windows, you can just use cheat engine to do it

**Comment by diogosilva212** [comment]
We're fuc*** up.

**Comment by ethano8225** [comment]
hold up if y’all have a thing called “wine” (wine is not an emulator, that’s the name) it may work because I think there’s cheat engine for Mac idk tho you may have to mess around with it

**Comment by ghost** [comment]
I know what wine is, it doesn't work on cheat engine. Wine is also suuuper buggy

**Comment by ethano8225** [comment]
oh lmao just an idea

do these pointers go up by like any # though? (Like can they not go up by 1 or something or can they go up by any #?

**Comment by ghost** [comment]
no thats not how it works at all

**Comment by ethano8225** [comment]
shit so they don’t go up by like 1 in hexadecimal or something?

**Comment by ghost** [comment]
no, its all dependent on what has changed in the dylib before it, if you added 154 bytes of code before the offset, then the next update it will be old off+ 154

**Comment by ethano8225** [comment]
oh shit then I don’t got (m)any ideas... the only thing that I can think of is that since there’s a new gamemode and like new models it souls be pushed a lot farther back I guess and that’s really all I can think of lol

**Comment by ghost** [comment]
hol up, there are 2 client.dylib's............. why valve

**Comment by amcgregor** [comment]
Disassembling the dylibs, unless you are a savant of X64 machine code, is akin to trying to find a lost contact lens in your front yard grass by cutting down each blade with an eyelash curler.  Wrong tool for the job, inappropriate scale of approach; the compiled code doesn't tell you the in-memory offsets, not directly, it describes how to calculate them… to the CPU.  An appropriate in-memory scanner like CheatEngine could work, but not with simulated system APIs via WINE; ye gods.  There are a number of Mac scanners advertised on unknowncheats, no?  (I know I've seen one mentioned by / bundled with other Github projects…)

@fugiefire Not entirely sure why you're asking, but type "man strstr" (without quotes) into Terminal.  (Fishing, not a fish. ;)

Edited because I nearly forgot.  Quote of the month, right here:

> shit so they don’t go up by like 1 in hexadecimal or something?

(For reference, the counting system in use only really matters if you have a number larger than the given quantity, 16 for hex.  E.g. 1 in decimal (normal) is 1 in hex is 1 in octal is 1 in binary… the difference is what the value of "10" is in each.)

**Comment by ethano8225** [comment]
> Disassembling the dylibs, unless you are a savant of X64 machine code, is akin to trying to find a lost contact lens in your front yard grass by cutting down each blade with an eyelash curler.

Hahahhahaahahah good analogy. good ol chuckle

**Comment by ghost** [comment]
eh its not that bad, and yeh i forgot that it would just show how to calc and not where it gets stored. I guess we could try gdb?... but gdb is eugh

**Comment by 7475** [comment]
Anyone have the new offsets yet?

**Comment by ghost** [comment]
we need the offsets!!!!

**Comment by 7475** [comment]
Surely someone must have it by now that is subscribed to the thread.  It would be awesome and we'd all be very thankful.

**Comment by macacodobem** [comment]
@diogosilva212 Did you have any news?

**Comment by imoplex** [comment]
The logic that previously worked for updating the offset (looking at the difference in offsets on PC and applying the same diff to our offset) doesn't seem to work this time. I think it's because all the offsets have changed in the new versions so we probably have to look at the other ones as well. I'll investigate more.

FYI, using the old logic of updating the offsets, the new glow index should be `0xACE8`
edit: Just saw that diogosilva212 said that as well

**Comment by diogosilva212** [comment]
@imoplex I tried that, yes. Didn't work tho.

**Comment by aKalisch** [comment]
DT_CSPlayer::m_flFlashDuration + 0x18

Get the netvar offset and add 0x18. Just as easy. But I told that before.

**Comment by diogosilva212** [comment]
@aKalisch how do we get the netvar offset? I'm sorry if it is a "noob" question, but I don't know...

**Comment by ghost** [comment]
lmao same i feel so dumb

**Comment by shaikhabrez** [comment]
nop its not working bro

**Comment by ghost** [comment]
all we need to do is convert 0xA3F8 to mac

**Comment by diogosilva212** [comment]
I just want that whoever finds the correct one gives us a tutorial on how to do it, so we could stop asking all these people...

**Comment by aKalisch** [comment]
Google how to find netvar offsets in the csgo libraries ;-)
It's pretty simple.

**Comment by ghost** [comment]
thats what im trying to do rn

**Comment by aKalisch** [comment]
https://www.youtube.com/watch?v=QdG9-jxOf8Y

Don't focus on cheatengine.. Focus on the IDA part.

**Comment by diogosilva212** [comment]
You just gave me a tutorial for pc...

**Comment by aKalisch** [comment]
The IDA Parts is the same on mac

**Comment by diogosilva212** [comment]
This one doesn't work? https://www.youtube.com/watch?v=3HhD7WNzSno

**Comment by ghost** [comment]
could be imma try it

**Comment by ghost** [comment]
it does work but i don't have time to do it unfortunately but if someone would do it it would be awesome

**Comment by diogosilva212** [comment]
I couldn't do it. And I'm getting tired of all these people knowing and saying it's so simple and they don't help.

**Comment by ghost** [comment]
0xAC38 glows one of my ct teammates

**Comment by 7475** [comment]
43 glowed squeaky for me in cache this morning hahaha

**Comment by diogosilva212** [comment]
I can't use that tutorial because when I shoot the bot and then I search for "80" I get thousands of values lmao The guy in the video gets 5 xD

**Comment by ghost** [comment]
0xAC78 made a couple of doors glow

**Comment by diogosilva212** [comment]
Ok, now I understand what @aKalisch said. On PC the m_flFlashDuration is 0xA3E0 and if you add 0x18 you get the 0xA3F8 (new m_iGlowIndex for PC).

Thing is... How the hell do I know the m_flFlashDuration for Mac?

**Comment by diogosilva212** [comment]
Oh, and m_flFlashMaxAlpha +0x1C also gives you the m_iGlowIndex.

**Comment by ghost** [comment]
so is it possible to find the new glow index (offset)on mac w that?

**Comment by 7475** [comment]
I'm lost.  Was so much easier when using the pattern scanner.

**Comment by ghost** [comment]
why cant it be used now?

**Comment by 7475** [comment]
I used it the last time for the Halloween update, and the output didn't work.

**Comment by imoplex** [comment]
> Ok, now I understand what @aKalisch said. On PC the m_flFlashDuration is 0xA3E0 and if you add 0x18 you get the 0xA3F8 (new m_iGlowIndex for PC).
> 
> Thing is... How the hell do I know the m_flFlashDuration for Mac?

flash the bot instead of shooting? i would try that (no time or IDA installed to try this but i would expect this to work)

**Comment by diogosilva212** [comment]
I don't know how to do that, but people in here say it's so easy, why they don't take 20 minutes of their time and do that instead of writing lol.

**Comment by randomuser123123** [comment]
wait so does this hack work or is it dead forever

**Comment by ghost** [comment]
it will work once someone finds the iglowindex offset

**Comment by diogosilva212** [comment]
On "unknowncheats" nobody gives an answer... I give up.

**Comment by ghost** [comment]
na we still gotta keep trying bro

**Comment by 7475** [comment]
just tried using the pattern scanner again, and the output on terminal just gives:

bash-3.2# ./Patternscanner
Segmentation fault: 11
bash-3.2#

**Comment by diogosilva212** [comment]
Try how? I don't know how to find the offsets, the video with bitslicer doesn't work, pattern scanner doesn't work... The guys that know how to do, don't do...

**Comment by ghost** [comment]
well maybe they're trying right now to figure it out

**Comment by 7475** [comment]
Can someone else try it: https://www.youtube.com/watch?v=aP99mdDzwQk

**Comment by macacodobem** [comment]
relax my friends, sooner or later they'll find the puzzle. Time is the key

**Comment by diogosilva212** [comment]
@7475 doesn't work

**Comment by 7475** [comment]
Yep, wouldn't work for the Halloween update.  Was worth a shot though.

**Comment by aKalisch** [comment]
<img width="858" alt="bildschirmfoto 2018-12-07 um 18 49 16" src="https://user-images.githubusercontent.com/3403599/49664026-e91b2c80-fa50-11e8-9f41-5bdd39454ae4.png">
Go for it ;-)

**Comment by aKalisch** [comment]
Took me 1 minute to figure it out...

**Comment by diogosilva212** [comment]
So... 0xAD00 + 0x18?

**Comment by diogosilva212** [comment]
I tried 0xAD18, didn't work... I'm fuc**** dumb or what?

**Comment by 7475** [comment]
You and me both… I'm close to 40 years old though… that's my excuse :)

**Comment by shaikhabrez** [comment]
any luck guyz ?

**Comment by diogosilva212** [comment]
More than one hundred comments full of nothing.

If it's so simple to him, I really don't understand why @aKalisch doesn't just throw the correct value of glowindex in here. Dude, I'm not trying to build a code, don't say one thing and leave, I just want to play a game.

This could be easier if you just explain how we do it for future ocasions and that's all. Jesus.

**Comment by diogosilva212** [comment]
More than one hundred comments full of nothing.

**Comment by jurupa** [comment]
I wish was only that. 
I tried already a couple variations.. including this one and nothing yet.
We hope someone can give us a help....

**Comment by shaikhabrez** [comment]
> Heyy so we only need to change
> 
> uint64_t m_iGlowIndex = 0xad00;
> bool statBool = true;
> 
> right? I'm not sure what to change the value to still. I'm on macOS.

if i drop my gun it glows loll by using this

**Comment by diogosilva212** [comment]
0xAD00 it's the m_flFlashDuration. m_iGlowIndex is that value + 0x18, that makes 0xAD18. Doesn't work.

**Comment by shaikhabrez** [comment]
random people is glowing by above 
uint64_t m_iGlowIndex = 0xad00;
sometime ct and sometime t

**Comment by alextechtai** [comment]
@diogosilva212 Is 0xeb91f0 the flash duration?

**Comment by aKalisch** [comment]
Then the signature might be outdated

**Comment by alextechtai** [comment]
@aKalisch What do you mean? How do you update the signature?

**Comment by diogosilva212** [comment]
Can you fix it? @aKalisch

**Comment by ethano8225** [comment]
hold up Kalish howd u finding that list because we could like maybe use that to help possibly (like the photo you got)

**Comment by diogosilva212** [comment]
@aKalisch this's the part of the code that maybe's broken?

```
LocalPlayerBase = clientScanner->getPointer(
        (Byte*)"\x89\xD6\x41\x89\x00\x49\x89\x00\x48\x8B\x1D\x00\x00\x00\x00\x48\x85\xDB\x74\x00",
        "xxxx?xx?xxx????xxxx?",
        0xB
    ) + 0x4;
    
    playerBase = clientScanner->getPointer(
        (Byte*)"\x48\x8D\x1D\x00\x00\x00\x00\x48\x89\xDF\xE8\x00\x00\x00\x00\x48\x8D\x3D\x00\x00\x00\x00\x48\x8D\x15\x00\x00\x00\x00\x48\x89\xDE",
        "xxx????xxxx????xxx????xxx????xxx",
        0x3
    ) + 0x2C;
    
    glowInfoOffset = clientScanner->getPointer(
        (Byte*)"\x48\x8D\x3D\x00\x00\x00\x00\xE8\x00\x00\x00\x00\x85\xC0\x0F\x84\x00\x00\x00\x00\x48\xC7\x05\x00\x00\x00\x00\x00\x00\x00\x00\x48\x8D\x35\x00\x00\x00\x00",
        "xxx????x????xxxx????xxx????xxxxxxx????",
        0x22
    ) + 0x4;
```

**Comment by aKalisch** [comment]
@diogosilva212 I have my own cheat that needs some maintenance due to the latest CS:GO Update.

**Comment by diogosilva212** [comment]
@aKalisch Hmm... So can you just tell us how to update the signatures?

**Comment by ethano8225** [comment]
I mean if aKalisch is fixing his own cheat he will probably find the #s so just if you stop bothering him he'll find them quicker lol

**Comment by diogosilva212** [comment]
Yeah, because asking him some questions is stoping him from doing whatever he has to do lol.

**Comment by ethano8225** [comment]
Well it would as letting him concentrate on one task is better than explaining the thing that you are doing (to an inexperienced person) while trying to actually accomplish that task is going to be more difficult

**Comment by imoplex** [comment]
AKalisch was right. I was away so I couldn't test the solution.
1. New offset is `0xAD18`
2. What's broken now is the logic behind determining team number. I don't have a solution for this. To make it work without detecting teams (basically, both sides will glow), jut comment out or remove the following code
```
//        if (playerTeamNum == iTeamNum || playerTeamNum == 0) {
//            continue;
//        }
//
//        if (playerTeamNum == 0) {
//            continue;
//        }
```

You will end up with something like this in-game. I'll try and find the correct offsets for the team num but i can't promise anytihng.
![copy](https://user-images.githubusercontent.com/44415962/49677230-00244380-fa7e-11e8-8aa4-e57d9aad0b94.jpg)

**Comment by imoplex** [comment]
Enemy team glow can be fixed by adding `+ 0x4` to this line
`int health              = mem->read<int>(memoryAddress + 0x134 + 0x4);`

**Comment by imoplex** [comment]
Actually, ignore step 2 in my instructions up there. Just add `+ 0x4` to wherever `0x128` is mentioned and only enemy team is highlighted correctly.

**Comment by 7475** [comment]
YOU LEGEND!  

At first I replaced the 128 with + 0x4, until I realised you meant add it after the 128.  Thank you so much for your hard work man!

**Comment by 7475** [comment]
This is beautiful :)

**Comment by 7475** [comment]
Just a heads up, doesn't seem to work in Danger Zone.

**Comment by ethano8225** [comment]
thanks lmao u are genius

**Comment by jurupa** [comment]
Youre the man! Its almost there.... It works in the first round, but then stops glowing at following rounds... maybe it is missing some last offset....

**Comment by YeBoiZ** [comment]
It doesn't seem to be working for me? Any tips? 

Edit: I only see glimpse of people once in a while, I changed to the new offset but that didnt help.

**Comment by ethano8225** [comment]
there are 2 places where 0x134 is mentioned, add to both. change to 0xAD18. not much else

**Comment by diogosilva212** [comment]
Thank you, @imoplex. LEGEND!

**Comment by diogosilva212** [comment]
Wait, there's still a problem. The glow works but doesn't show the correct health of the player. It's always red :(

**Comment by ethano8225** [comment]
ehh, doesnt really matter as of now, we can deal with red instead of colors lol

**Comment by ethano8225** [comment]
but @imoplex do you get any type of explanation on how u just did that? because it may make it a bit easier so we dont have to tag you or anything

**Comment by macacodobem** [comment]
for me the adversary team appeared red and the friend appears green.

      42   uint64_t m_iGlowIndex       = 0xAD18;

      62    int health              = mem->read<int>(memoryAddress + 0x134);
      63    int playerTeamNum       = mem->read<int>(memoryAddress + 0x128 + 0x4);



Anyone know how to take the glow out of friend team?

**Comment by itsfrank115** [comment]
so is there a way the glow can display health or is it still a work in progress?

**Comment by ghost** [comment]
... i went to sleep and all this happened lmfao

**Comment by ghost** [comment]
If you wanna display health make line 62:
`int health              = mem->read<int>(memoryAddress + 0x134+4);`

**Comment by 7475** [comment]
Does anyone know why it won't work in Dangerzone?

**Comment by jefcostaa** [comment]
if (playerTeamNum == iTeamNum) {
        Color color = {float((100 - health) / 200.0), float((health) / 100.0), (255 / 255), 0.5f};
 
Maybe work in dangerzone

**Comment by jefcostaa** [comment]
i if (playerTeamNum == iTeamNum) {
        Color color = {float((100 - health) / 200.0), float((health) / 100.0), (255 / 255), 0.5f};
        continue;
    }
    if (playerTeamNum == 0) {
        continue;
    }

**Comment by 7475** [comment]
Did it work for you?

**Comment by imoplex** [comment]
> but @imoplex do you get any type of explanation on how u just did that? because it may make it a bit easier so we dont have to tag you or anything

No problem. The hard part was by @aKalisch so full props to him here. I think we should listen to him and learn how to use IDA if we want this maintained.

In the glow we have, we use a several offsets, `m_iGlowIndex` which changes with most major updates, `m_iHealth` and `m_iTeamNum` which have historically stayed the same (at least since i started looking at this).

Usually, i would just take a look at the changes in offsets from the PC version (https://github.com/frk1/hazedumper) and apply the same difference to our `m_iGlowIndex` and all would be fine but with the last big update not only was this difference not the same between PC and Mac but the other offsets have changed. Thankfully, once @aKalisch gave the correct offset for the glow, it was just a matter of looking at the diffs to the other offsets to make them work. In my first post i just removed all of the code that related to team num and health logic so we could test whether glow was working (which is why all players were glowing and incorrectly colored).

**Comment by ghost** [comment]
is there a separte offset for danger zone mode?

**Comment by ghost** [comment]
@jefcostaa did u get it to work for danger zone?

**Comment by jefcostaa** [comment]
<img width="787" alt="captura de tela 2018-12-08 as 23 06 46" src="https://user-images.githubusercontent.com/44112476/49692285-f4985180-fb3d-11e8-87e3-01fede718bf3.png">

**Comment by sexypoop192** [comment]
can someone send the full code with lines 42,62,63 please

**Comment by ethano8225** [comment]
@BotJake69 ill just send u a link of main.cpp one second (on pastebin, youll have to copy and paste it into your own main.cpp

**Comment by ethano8225** [comment]
alright, its here @BotJake69 https://pastebin.com/eY4mTFva

**Comment by sexypoop192** [comment]
@etan8225 thank you so much man!

**Comment by alextechtai** [comment]
@etan8225 The code somehow doesn't show low health anymore?

**Comment by imoplex** [comment]
0x134 should be 0x138 on the int health line, said it before already

**Comment by ghost** [comment]
@jefcostaa can you send me the full code plz, im rly stupid.

**Comment by 7475** [comment]
@jefcostaa  Hey man, I copied and pasted lines 65-71, and still doesn't work in  Danger Zone.  Comes up as unused variable, Color etc.  Have you gotten it to work in Danger Zone?

**Comment by amcgregor** [comment]
Some reading:

* [GitHub Flow](https://help.github.com/articles/github-flow/)
* [Proposing Changes with Pull Requests](https://help.github.com/articles/proposing-changes-to-your-work-with-pull-requests/)

Given there are [already two](
https://github.com/iseekwonderful/csgoGlow/pull/267) PRs [for this problem](
https://github.com/iseekwonderful/csgoGlow/pull/266), has anyone tried one of those forks/branches out?

Seeing code, or hell, even just static integers, exchanged in comments, or comments requesting "the full code"… on a code sharing and hosting site… seems silly to me; like intentionally picking the least efficient method of communication.  (Insert Mars Attacks aliens "wark-ing" here.)  This site actively encourages forks and true collaborative development.  Not this play-by-email approach with each interested person manually applying a change-by-comment, hopefully the same as everyone else.

@MagzM https://github.com/jefcostaa/csgoGlow — clicked his name, there it was, here's hoping it's the one he's modifying.  See the links I gave for an explanation of how GitHub usually works, re: forks.

**Comment by sexypoop192** [comment]
Whenever I put it in terminal i get 
Found CSGO PID: 1486
Found the Client.dylib address: 0x1226c8000 
Module should end at 0x123bb137c
@etan8225

**Comment by jefcostaa** [comment]
https://github.com/jefcostaa/csgoGlow 

For those with difficulty, here is the complete file to use.

It will only work on DangerZone, so to use it in competitive I recommend you leave another file without these modifications just for the comp.

The only difference of this modified file that I created is that you can only drag the "wall" into the terminal after you get off the Helicopter, on the contrary the game will lock in the heat.

Enjoy,

**Comment by alextechtai** [comment]
@jefcostaa What does that last paragraph mean? 😅

**Comment by ethano8225** [comment]
> @jefcostaa What does that last paragraph mean? 😅

basically you have to inject as soon as you drop out of the helicopter or something like your game will freeze or it just wont work.

**Comment by ghost** [comment]
@etan8225 why does it freeze if you inject before dropping off heli? It doesn't happen on mine

**Comment by rpranshu** [comment]
![screen shot 2018-12-25 at 8 24 12 pm](https://user-images.githubusercontent.com/16818486/50424046-608bf100-0884-11e9-9141-44c8d6a7fd45.png)

Is this fine? @fugiefire

**Comment by etan8225** [comment]
> @etan8225 why does it freeze if you inject before dropping off heli? It doesn't happen on mine

im rertarded it work perfectly and

> ![screen shot 2018-12-25 at 8 24 12 pm](https://user-images.githubusercontent.com/16818486/50424046-608bf100-0884-11e9-9141-44c8d6a7fd45.png)
> 
> Is this fine? @fugiefire

no change line 42 to 0xAD18 and i think thats it (i just ate deoderant and feel like shit for a dare ofc)

**Comment by 7475** [comment]
Quick question, does anyone know if it is working in Mojave.  Must update my system :(

**Comment by ethano8225** [comment]
@7475 yes it does, you just have to enable terminal/the csgoGlow terminal app to control the computer in preferences (in security and privacy).
